rice-middleware/kns/src/main/java/org/kuali/rice/kns/web/struts/action/KualiDocumentActionBase.java

```diff
@@ -818,7 +800,8 @@ public ActionForward route(ActionMapping mapping, ActionForm form, HttpServletRe
             return forward;
         }
 
-        document = getDocumentService().routeDocument(document, kualiDocumentFormBase.getAnnotation(), combineAdHocRecipients(kualiDocumentFormBase));
+        String nodeName = document.getAdHocRouteNodeName();
+        document = getDocumentService().routeDocument(document, kualiDocumentFormBase.getAnnotation(), nodeName, combineAdHocRecipients(kualiDocumentFormBase));
```

Comment:
I don't think you need to do this, just call the version of routedocument without passing the node name, it's going to do the same thing you've done manually above by invoking document.getAdHocRouteNodeName

This applies in the other parts of this file as well, I don't think you even need to change KualiDocumentActionBase. That keeps it a little bit simpler and cleaner in my mind.